controllers/bindings/errors.go

```diff
@@ -16,6 +16,23 @@ package bindings
 
 import "errors"
 
+type ErrorReason string
+
+const (
+	ErrorReasonNone ErrorReason = ""
+
+	// XXX: note that this used to be used as:
+	// - api.SPIAccessTokenBindingErrorReasonTokenSync originally in secretHandler.Sync
+	ErrorReasonSecretUpdate ErrorReason = "SecretUpdate"
+	// XXX: note that this used to be used as:
+	// - api.SPIAccessTokenBindingErrorReasonServiceAccountUnavailable in ensureReferencedServiceAccount -> serviceAccountHandler.Sync
+	ErrorReasonServiceAccountUnavailable ErrorReason = "ServiceAccountUnavailable"
+	// XXX: note that this used to be used as:
+	// - api.SPIAccessTokenBindingErrorReasonServiceAccountUpdate in ensureReferencedServiceAccount -> serviceAccountHandler.Sync
+	// - api.SPIAccessTokenBindingErrorReasonTokenSync in ensureReferencedServiceAccount -> serviceAccountHandler.Sync
```

Comment:
We always took great care to list all the error reasons as the values of the appropriate type in the CRD. In this MVP I forgave that in the name of simplicity and am simply forcefully converting a string into the error reason type mentioned in the CRD. I think it would be good to go back to using the strongly typed error reasons and therefore left a note here in the code.

To answer your supposed follow up question with a question... Is it right to create an issue for fixing something that is not yet in the main branch?